independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java

```diff
@@ -136,10 +143,30 @@ private LocalProject(Model rawModel, LocalWorkspace workspace) throws BootstrapE
         this.workspace = workspace;
         this.groupId = ModelUtils.getGroupId(rawModel);
         this.artifactId = rawModel.getArtifactId();
-        this.version = ModelUtils.getVersion(rawModel);
+
+        String version = ModelUtils.getVersion(rawModel);
+        final boolean revisionVersion;
+        if (revisionVersion = REVISION_EXPR.equals(version)) {
+            version = rawModel.getProperties().getProperty(REVISION);
```

Comment:
Does this also cover `-Drevision=<something_different_than_in_pom>`?
This is important, because that is what CI (e.g. Jenkins via `Jenkinsfile`) would be passing to `mvn` to override the default revision.

A variation is of that would be:
- ```xml
  <properties>
      <revision>0.1-${revisionSuffix}</revision>
      <revisionSuffix>SNAPSHOT</revisionSuffix>
  </properties>
  ```
- CI passes `-DrevisionSuffix=build-${BUILD_NUMBER}`

This way local build produces `0.1-SNAPSHOT` and CI produces something like `0.1-build-123`.

Reply:
Actually according to the Maven doc you referenced this is not supported by Maven itself, isn't it?

Reply:
I tried adding support for sha1 and changelist but strangely it also didn't work in Maven itself. Unless I somehow did it wrong.

Reply:
> Does this also cover `-Drevision=<something_different_than_in_pom>`?

This will work though.

Reply:
> Actually according to the Maven doc you referenced this is not supported by Maven itself, isn't it?

Which part do you mean? `-Drevision=...` is mentioned explicitly multiple times.

The `revisionSuffix` is not mentioned but IMHO this is just basic Maven interpolation logic, no?
You could also have `myFancySomethingProperty` instead of `revisionSuffix`, doesn't matter.

> I tried adding support for sha1 and changelist but strangely it also didn't work in Maven itself.

I swapped `revision` with `sha1` locally in my sample project and it worked (apart from the Quarkus build problem). I even changed the default value from `1.0-SNAPSHOT` to `0.2-SNAPSHOT` and I also tried `-Drevision=1.1-SNAPSHOT`.

> > Does this also cover `-Drevision=<something_different_than_in_pom>`?
> 
> This will work though.

Ok, as long as this works the main problem is solved. The rest could be adressed separately.

Reply:
> You could also have `myFancySomethingProperty` instead of `revisionSuffix`, doesn't matter.

Arbitrary property interpolation is not allowed in the version elements. There is an example in that doc.

> I swapped `revision` with `sha1` locally in my sample project and it worked

Ah, I need to try that. I tried combining the three mentioned properties like they had in the doc, which for some reason didn't work.

Reply:
> > You could also have `myFancySomethingProperty` instead of `revisionSuffix`, doesn't matter.
> 
> Arbitrary property interpolation is not allowed in the version elements. There is an example in that doc.

You are right, I have totally missed that!

> > I swapped `revision` with `sha1` locally in my sample project and it worked
> 
> Ah, I need to try that. I tried combining the three mentioned properties like they had in the doc, which for some reason didn't work.

I just tried the following and everything works as expected (in Maven itself):
```diff
$ git diff
diff --git a/core/pom.xml b/core/pom.xml
index ea45976..b52f070 100644
--- a/core/pom.xml
+++ b/core/pom.xml
@@ -5,7 +5,7 @@
   <parent>
     <groupId>com.github.famod.modmono-quarkus</groupId>
     <artifactId>modmono-quarkus-parent</artifactId>
-    <version>${revision}</version>
+    <version>${revision}${sha1}</version>
   </parent>
   <artifactId>modmono-quarkus-core</artifactId>
   <dependencies>
diff --git a/dist/pom.xml b/dist/pom.xml
index 278e9c5..e0df457 100644
--- a/dist/pom.xml
+++ b/dist/pom.xml
@@ -5,7 +5,7 @@
   <parent>
     <groupId>com.github.famod.modmono-quarkus</groupId>
     <artifactId>modmono-quarkus-parent</artifactId>
-    <version>${revision}</version>
+    <version>${revision}${sha1}</version>
   </parent>
   <artifactId>modmono-quarkus-dist</artifactId>
   <dependencies>
diff --git a/pom.xml b/pom.xml
index b895c66..f72103c 100644
--- a/pom.xml
+++ b/pom.xml
@@ -4,10 +4,11 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.github.famod.modmono-quarkus</groupId>
   <artifactId>modmono-quarkus-parent</artifactId>
-  <version>${revision}</version>
+  <version>${revision}${sha1}</version>
   <packaging>pom</packaging>
   <properties>
-    <revision>1.0-SNAPSHOT</revision>
+    <revision>0.3</revision>
+    <sha1>-SNAPSHOT</sha1>

     <compiler-plugin.version>3.8.1</compiler-plugin.version>
```